File: myfragments/src/main/java/pl/mareklangiewicz/myfragments/MyDecoratorsTestsFragment.java
package pl.mareklangiewicz.myfragments;


import android.os.Bundle;
import android.support.annotation.IdRes;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import pl.mareklangiewicz.myviews.MyViewDecorator;


public final class MyDecoratorsTestsFragment extends MyFragment {

    public MyDecoratorsTestsFragment() {
    }

    @Override public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        super.onCreateView(inflater, container, savedInstanceState); //just for logging

        inflateHeader(R.layout.mf_my_basic_header);
        setHasOptionsMenu(true);
        return inflater.inflate(R.layout.mf_my_decorators_tests_fragment, container, false);
    }

    @Override public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        inflater.inflate(R.menu.mf_my_decorators, menu);
    }

    @Override public boolean onOptionsItemSelected(MenuItem item) {
        @IdRes int id = item.getItemId();
        if(id == R.id.action_decorate_views) {
            View view = getView();
            if(view == null) {
                log.e("The root view of this fragment is null.");
                return false;
            }
            MyViewDecorator.decorateTree(view, "decorate", R.layout.mv_example_decoration, null);
            return true;
        }
        return false;
    }
}

File: myfragments/src/main/java/pl/mareklangiewicz/myfragments/MyDecoratorsTestsFragment.kt
package pl.mareklangiewicz.myfragments


import android.os.Bundle
import android.view.*
import pl.mareklangiewicz.myutils.e
import pl.mareklangiewicz.myviews.MyViewDecorator


open class MyDecoratorsTestsFragment : MyFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        super.onCreateView(inflater, container, savedInstanceState) //just for logging

        manager?.lnav?.headerId = R.layout.mf_my_basic_header
        setHasOptionsMenu(true)
        return inflater.inflate(R.layout.mf_my_decorators_tests_fragment, container, false)
    }

    override fun onDestroyView() {
        manager?.lnav?.headerId = -1
        super.onDestroyView()
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) = inflater.inflate(R.menu.mf_my_decorators, menu)

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId){
            R.id.action_decorate_views -> {
                val view = view
                if (view == null) {
                    log.e("The root view of this fragment is null.")
                    return false
                }
                MyViewDecorator.decorateTree(view, "decorate", R.layout.mv_example_decoration)
                return true
            }
        }
        return false
    }
}
